### PR TITLE
Frontend uc300 create customer

### PR DESCRIPTION
**Pull request by gubelli**
Version ready on staging

### CODE REVIEW

File: Frontend/src/app/customer/customer-create.component.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import {Component} from '@angular/core';
+import {Customer} from './customer.type';
+import {CustomerService} from '../services/customer.service';
+import { Location } from '@angular/common';
+import {SnackbarService} from '../services/snackbar.service';
+import 'rxjs/add/operator/switchMap';
+import {NgForm} from "@angular/forms";
```
Comment (Alucard1766) [inline Frontend/src/app/customer/customer-create.component.ts]: ' anstelle von "

File: Frontend/src/app/customer/customer-detail.component.ts [hunk 2]
```diff
@@ -12,16 +12,33 @@ import 'rxjs/add/operator/switchMap';
...
       .switchMap((params: ParamMap) => this.customerService.getCustomer(+params.get('id')))
-      .subscribe(customer => this.customer = customer);
+      .subscribe(customer => this.customer = customer['result'],
+        error => this.snackBar.notifyError(error));
+  }
+
+  saveCustomer(): void {
+    console.log(this.customer);
+    this.customerService.putCustomer(this.customer).subscribe( data => {
+      this.snackBar.notifySuccess('Modification successful saved')
```
Comment (Alucard1766) [inline Frontend/src/app/customer/customer-detail.component.ts]: Semicolon missing after notifySuccess()

File: Frontend/src/app/customer/customer.component.css [hunk 3]
```diff
@@ -9,23 +10,23 @@ header > mat-toolbar{
...
-  font-size: 20px;
+@media only screen and (max-device-height: 800px) {
+  .mat-table {
+    max-height: 300px;
+  }
+}
+
+@media only screen and (min-device-height: 801px) {
+  .mat-table {
+    max-height: 500px;
```
Comment (Alucard1766) [inline Frontend/src/app/customer/customer.component.css]: Z18/Z24: Mit % möglich anstelle von fixem Wert?

File: Frontend/src/app/customer/customer.component.ts [hunk 4]
```diff
@@ -47,7 +47,7 @@ export class CustomerTableSource extends DataSource<any> {
   connect(): Observable<Customer[]> {
```
Comment (Alucard1766) [inline Frontend/src/app/customer/customer.component.ts]: Do not export class CustomerTableSource

File: Frontend/src/app/customer/customer-detail.component.html [hunk 5]
```diff
@@ -6,25 +6,40 @@
...
       </mat-form-field>
       <mat-form-field fxFlex="grow">
-        <input matInput placeholder="Phone Number" #customerPhone>
+        <input matInput [disabled]="!editable" placeholder="Phone number" name="phoneNumber" [(ngModel)]="customer.phoneNumber">
       </mat-form-field>
     </form>
-    <div class="sub-button">
+    <div>
       <button mat-raised-button color="accent" (click)="goBack()">Back</button>
+      <button *ngIf="editable" [disabled]="!customerForm.valid||!customerForm.dirty" mat-raised-button color="accent" (click)="saveCustomer(customerForm)">Save</button>
```
Comment (Alucard1766) [inline Frontend/src/app/customer/customer-detail.component.html]: Move Buttons to top

File: Frontend/src/app/services/snackbar.service.ts [hunk 6]
```diff
@@ -1,16 +1,20 @@
...
+import {ErrorSnackbarComponent} from '../snackbars/error.snackbar';
 
 @Injectable()
 export class SnackbarService{
-  constructor(private snackbar: MatSnackBar){}
+  constructor(private snackbar: MatSnackBar) { }
 
-  notify(errorMessage: any): void {
+  notifyError(errorMessage: any): void {
     this.snackbar.openFromComponent(ErrorSnackbarComponent, {
```
Comment (Alucard1766) [inline Frontend/src/app/services/snackbar.service.ts]: Rename snackbars:
image
to error.snackbar.component.[css|html|ts]

File: Frontend/src/app/customer/customer-detail.component.css [hunk 7]
```diff
@@ -1,6 +1,8 @@
 :host{
```
[the remark this answers is not in the document]
Comment (gubelli) [reply Frontend/src/app/customer/customer-detail.component.css]: font-color: black

**Pull request merged**